cli/src/main/java/com/devonfw/tools/ide/tool/LocalToolCommandlet.java

```diff
@@ -75,18 +75,13 @@ public final boolean install(boolean silent, EnvironmentContext environmentConte
     VersionIdentifier installedVersion = getInstalledVersion();
     Step step = this.context.newStep(silent, "Install " + this.tool, configuredVersion);
     try {
-      // TODO https://github.com/devonfw/IDEasy/issues/664
-      boolean enableOptimization = false;
-      // performance: avoid calling installTool if already up-to-date
-      if (enableOptimization & configuredVersion.equals(installedVersion)) { // here we can add https://github.com/devonfw/IDEasy/issues/637
-        return toolAlreadyInstalled(silent, installedVersion, step);
-      }
       // install configured version of our tool in the software repository if not already installed
       ToolInstallation installation = installTool(configuredVersion, environmentContext);
 
       // check if we already have this version installed (linked) locally in IDE_HOME/software
       VersionIdentifier resolvedVersion = installation.resolvedVersion();
-      if (resolvedVersion.equals(installedVersion) && !installation.newInstallation()) {
+      if (resolvedVersion.equals(installedVersion) && !installation.newInstallation()
+          || configuredVersion.matches(installedVersion) & context.isSkipUpdatesMode()) {
```

Comment:
I do not know why Java ever allowed this syntax but by convention `&&` or `||` should always be used for logical AND or OR while `&` and `|` should be used for bit-wise `AND` or `OR`. It turns out that for `boolean` values the compiler accepts both variants and produces the same result but IMHO it is confusing and raises the question if the developer did some magic trick by first using `&&` and later using `&` that the reader may not understand. Also even though not required due to operator priority, it IMHO improves readability to explicitly use parenthesis to express how this combination of expressions will be evaluated (reader could even think left-to-right what would be very wrong).
```suggestion
      if ((resolvedVersion.equals(installedVersion) && !installation.newInstallation())
          || (configuredVersion.matches(installedVersion) && context.isSkipUpdatesMode())) {
```